core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -245,18 +247,20 @@ class ReplicaManagerTest {
       
       assertTrue(fetchCallbackFired)
       assertEquals("Should not give an exception", Errors.NONE.code, fetchError)
+      assertNotEquals("Should not return empty response", MessageSet.Empty, fetchedMessages)
```

Comment:
Would it not be better to do something like `assertTrue(fetchedMessages.iterator.hasNext)`? That would also catch the case where we return an empty response but without using `MessageSet.Empty`.